dbms/src/Flash/Pipeline/Exec/PipelineExec.cpp

```diff
@@ -107,16 +135,23 @@ OperatorStatus PipelineExec::awaitImpl()
 {
     auto op_status = sink_op->await();
     if (op_status != OperatorStatus::NEED_INPUT)
+    {
+        setIOOpIfNeeded(op_status, sink_op);
         return op_status;
-    for (auto it = transform_ops.rbegin(); it != transform_ops.rend(); ++it)
+    }
+    for (auto it = transform_ops.rbegin(); it != transform_ops.rend(); ++it) // NOLINT(modernize-loop-convert)
     {
         // If the transform_op returns `NEED_INPUT`,
         // we need to call the upstream transform_op until a transform_op returns something other than `NEED_INPUT`.
         op_status = (*it)->await();
         if (op_status != OperatorStatus::NEED_INPUT)
+        {
+            setIOOpIfNeeded(op_status, (*it));
             return op_status;
+        }
     }
     op_status = source_op->await();
+    setIOOpIfNeeded(op_status, source_op);
```

Comment:
There is so many 
```
setIOOpIfNeeded(op_status, (*it));
return op_status;
```
Maybe we can use a new function to do it?

Reply:
ok, updated.